File: android/app/src/main/java/it/curzel/bitscape/designsystem/Typography.swift
import Foundation
import SwiftUI

extension View {
    func typography(_ font: AppFont) -> some View {
        self
            .font(font.font)
            .lineSpacing(font.lineSpacing)
    }
}

struct AppFont {
    let size: CGFloat
    let name: String
    let weight: Font.Weight
    let font: Font
    let lineSpacing: CGFloat

    init(name: String, size: CGFloat, weight: Font.Weight) {
        self.name = name
        self.size = size
        self.weight = weight
        self.font = .custom(name, size: size).weight(weight)
        self.lineSpacing = size / 3
    }
}

extension AppFont {
    static let largeTitle = AppFont(name: "PixelOperator8-Bold", size: 24, weight: .bold)
    static let title = AppFont(name: "PixelOperator8-Bold", size: 16, weight: .bold)
    static let text = AppFont(name: "PixelOperator8", size: 14, weight: .regular)
    static let menuOption = AppFont(name: "PixelOperator8", size: 16, weight: .regular)
    static let caption = AppFont(name: "PixelOperator8", size: 11, weight: .regular)
}

File: android/app/src/main/java/it/curzel/bitscape/designsystem/Typography.kt
package it.curzel.bitscape.designsystem

import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import it.curzel.bitscape.R

object Typography {
    val largeTitle = TextStyle(fontFamily = PixelOperator8, fontWeight = FontWeight.Bold, fontSize = 24.sp)
    val title = TextStyle(fontFamily = PixelOperator8, fontWeight = FontWeight.Bold, fontSize = 16.sp)
    val text = TextStyle(fontFamily = PixelOperator8, fontWeight = FontWeight.Normal, fontSize = 14.sp)
    val menuOption = TextStyle(fontFamily = PixelOperator8, fontWeight = FontWeight.Normal, fontSize = 16.sp)
    val caption = TextStyle(fontFamily = PixelOperator8, fontWeight = FontWeight.Normal, fontSize = 11.sp)
}

private val PixelOperator8 = FontFamily(
    Font(R.font.pixeloperator8, FontWeight.Normal),
    Font(R.font.pixeloperator8, FontWeight.SemiBold),
    Font(R.font.pixeloperator8_bold, FontWeight.Bold),
    Font(R.font.pixeloperator8, FontWeight.Light)
)
